File: frontend/frontend/email-sorter/src/app/services/email.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Email {
  id: number;
  gmail_id: string;
  subject: string;
  sender: string;
  content: string;
  summary?: string;
  received_at: string;
  category_id?: number;
  gmail_account_id: number;
  is_archived: boolean;
  gmail_account?: {
    email: string;
  };
}

@Injectable({
  providedIn: 'root'
})
export class EmailService {
  private apiUrl = 'http://localhost:8000/api/v1';

  constructor(private http: HttpClient) {}

  getEmails(categoryId?: number, search?: string, skip = 0, limit = 50): Observable<Email[]> {
    let params = new HttpParams()
      .set('skip', skip.toString())
      .set('limit', limit.toString());

    if (categoryId) {
      params = params.set('category_id', categoryId.toString());
    }

    if (search) {
      params = params.set('search', search);
    }

    return this.http.get<Email[]>(`${this.apiUrl}/emails`, { params });
  }

  getEmail(id: number): Observable<Email> {
    return this.http.get<Email>(`${this.apiUrl}/emails/${id}`);
  }

  bulkDeleteEmails(emailIds: number[]): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/emails/bulk-delete`, emailIds);
  }
}

File: backend/app/schemas/email.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional
from .gmail_account import GmailAccount

class EmailBase(BaseModel):
    gmail_id: str
    subject: str
    sender: str
    content: str
    summary: Optional[str] = None
    received_at: datetime

class EmailCreate(EmailBase):
    category_id: Optional[int] = None

class EmailUpdate(BaseModel):
    category_id: Optional[int] = None
    summary: Optional[str] = None
    is_archived: Optional[bool] = None

class Email(EmailBase):
    id: int
    category_id: Optional[int] = None
    user_id: int
    gmail_account_id: int
    is_archived: bool
    created_at: datetime
    updated_at: datetime
    gmail_account: Optional[GmailAccount] = None

    class Config:
        from_attributes = True